ES6-modularization

Jmiccolo opened the pull request:
***<ins>Issue</ins>***
Use ES6 modules to modularise JavaScript

***<ins>What has Changed</ins>***
-Index.js has been reduced to a single line of code to initialize the application
-Created folders based on function 
-Code has been split into multiple files to maintain readability and function
-firebase_quill_init.js renamed quill_init.js to avoid confusion

***<ins>Folders</ins>****

1.  /init
- avatars.js  - a single exported variable of the avatar list
- init.js - exports default function to initialize application
- setUser.js - sets or gets user from Local Storage initialized in init()
- showContainers - removes "hidden" class from containers initialized in init()
2. /events
- eventHandlers.js - moves handlers for event listeners to separate file
- eventListeners.js - moves all event listeners to a single instance to be initialized upon init()
- serviceWorker.js - moves serviceWorker handling to events folder
3. /lessons
- helper.js - moves helper function to own file for creation of lessons
- index.js - leaves all lesson functions in single index file, might need to be split more
4. /firebase
 -commented out firebase.js to be removed

jrobind commented on app-main/public/js/events/eventHandlers.js:
```diff
@@ -0,0 +1,54 @@
+const submitLessonElement = document.querySelector("#submit");
+const clearBtn = document.querySelector(".create-lesson-clear");
+
+// Remove text from form
```
nitpick: I would say this comment isn't necessary. It's also a little misleading seeing as the function also handles the setting of textContent and removing attributes.

We can leave it as-is, just thought I'd mention it going forward.

jrobind replied:
My approach to commenting is:

If the code itself can be easily read and understood using appropriate variable names, methods, etc, then it's fine without comments. However, If there's unusual behaviour that's difficult to reason about, I'd write some.

jrobind commented on app-main/public/js/events/eventHandlers.js:
```diff
@@ -0,0 +1,54 @@
+  }
+}
+
+// close modal
```
nitpick: same as above re, comments. It's clear what this handles from the function name.

jrobind commented on app-main/public/js/events/serviceWorker.js:
```diff
@@ -0,0 +1,13 @@
+export default async function registerSW() {
+  if ("serviceWorker" in navigator) {
+    try {
+      await navigator.serviceWorker.register("../../sw.js");
```
Looks like we have some service worker caching issues. This will be due to the path change. I'd recommend moving all of the service worker registration logic back to `index.js` and reverting the registration path. You can see the errors in Chrome DevTools by toggling the network to offline and refreshing the page. Because the JS isn't cached the app fails to load.

![Screenshot 2021-01-07 at 10 51 11](https://user-images.githubusercontent.com/30320536/103885690-4bcfce00-50d8-11eb-9a00-c537531a19be.png)

jrobind reviewed (commented):
Thanks for this @Jmiccolo. This is looking really good – much tidier now 😄 

Just a few things that need fixing:

1. Service worker caching (left you a comment regarding this)
2. The tagging functionality looks like it might be broken too?

**Live**:
![Screenshot 2021-01-07 at 11 03 35](https://user-images.githubusercontent.com/30320536/103887733-7d966400-50db-11eb-83a5-eb3e59395ff2.png)

**This PR**:
![Screenshot 2021-01-07 at 11 30 37](https://user-images.githubusercontent.com/30320536/103887946-d1a14880-50db-11eb-88fd-24c62b02e932.png)

Jmiccolo commented:
So I think I fixed the tag issues, wasn't up to date in my local repository when I finished up so I didn't see the commit four days ago. I think I fixed what was necessary there. 

I don't know what I am doing wrong that I can't recreate those caching errors though. All of the files from the sw.js CACHE_FILES are caching for me. I don't know if you can explain any differently how to recreate those errors.

Jmiccolo commented:
Here is what my cache looks like after loading the page. 
![image](https://user-images.githubusercontent.com/65175521/103926077-7032a700-50e6-11eb-9b44-bede79835a19.png)

jrobind commented:
> So I think I fixed the tag issues, wasn't up to date in my local repository when I finished up so I didn't see the commit four days ago. I think I fixed what was necessary there.
> 
> I don't know what I am doing wrong that I can't recreate those caching errors though. All of the files from the sw.js CACHE_FILES are caching for me. I don't know if you can explain any differently how to recreate those errors.

Just checked your latest changes locally and the tagging issues are fixed, thanks! 

Regarding the sw caching errors, I've tested this again and it's working fine for me too. Let's get this merged in, if there are any issues we can always raise another ticket to address them.

Thanks again!
